packages/tools/src/tools/ReferenceCursors.ts

```diff
@@ -0,0 +1,471 @@
+  VolumeViewport,
+  utilities,
+  getEnabledElementByIds,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import {
+  addAnnotation,
+  getAnnotations,
+} from '../stateManagement/annotation/annotationState';
+import { isAnnotationVisible } from '../stateManagement/annotation/annotationVisibility';
+import { drawLine } from '../drawingSvg';
+import { getViewportIdsWithToolToRender } from '../utilities/viewportFilters';
+import {
+  EventTypes,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+  Annotation,
+  Annotations,
+} from '../types';
+import { ReferenceCursor } from '../types/ToolSpecificAnnotationTypes';
+
+import triggerAnnotationRenderForViewportIds from '../utilities/triggerAnnotationRenderForViewportIds';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+import { vec3 } from 'gl-matrix';
+import AnnotationDisplayTool from './base/AnnotationDisplayTool';
+import vtkMath from '@kitware/vtk.js/Common/Core/Math';
+import {
+  hideElementCursor,
+  resetElementCursor,
+} from '../cursors/elementCursor';
+import { getToolGroup } from '../store/ToolGroupManager';
+
+/**
+ * CursorCrosshairSyncTool is a tool that will show your cursors position in all other elements in the toolGroup if they have a matching FrameOfReference relative to its position in world space.
+ * Also when positionSync is enabled, it will try to sync viewports so that the cursor can be displayed in the correct position in all viewports.
+ *
+ * Configuration:
+ * - positionSync: boolean, if true, it will try to sync viewports so that the cursor can be displayed in the correct position in all viewports.
+ * - disableCursor: boolean, if true, it will hide the cursor in all viewports. You need to disable and reactivate the tool for this to apply.
+ * - displayThreshold: number, if the distance of the cursor in a viewport is bigger than this threshold the cursor will not be displayed.
+ *
+ * Only uses Active and Disabled state
+ */
+class ReferenceCursors extends AnnotationDisplayTool {
+  static toolName;
+  touchDragCallback: any;
+  mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  isDrawing = false;
+  isHandleOutsideImage = false;
+  _elementWithCursor: null | HTMLDivElement = null;
+  _currentCursorWorldPosition: null | Types.Point3 = null;
+  _currentCanvasPosition: null | Types.Point2 = null;
+  //need to keep track if this was enabled when tool was enabled because we need to know if we should reset cursors
+  _disableCursorEnabled = false;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        shadow: true,
+        preventHandleOutsideImage: false,
+        displayThreshold: 5,
+        positionSync: true,
+        disableCursor: false,
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+    this._disableCursorEnabled = this.configuration.disableCursor;
```

Comment:
This might cause a bug when configuration changes, since you are only assigning it on the construtor, maybe always use `this.configuration.disableCursor`?